### PR TITLE
feat: Authentication

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
1. login
2. signUp
3. logOut

Screenshot (74)
Screenshot (75)
Screenshot (76)

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { createContext, useContext, useState } from "react";
+import { loginServices, signUpServices } from "../services/Services";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const localStorageToken = JSON.parse(localStorage.getItem("auth"));
+  const [token, setToken] = useState(localStorageToken?.token);
```
Comment (ramalingaa) [inline src/context/auth-context.js:9]: This is a cool way to check the object values

**Review by ramalingaa — APPROVED** [review]
Clean and Modular code, Good logic implementaion. PR can be merged

File: src/pages/Auth/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+import { useAuth } from "../../context/auth-context";
+import "./Auth.css"
+import { useNavigate } from "react-router-dom";
+
+
+export const Login=()=>{
+    const [loginForm,setLoginForm]= useState({email:"",password:""});
+    console.log(loginForm);
```
Comment (rohit8686) [inline src/pages/Auth/Login.jsx]: better to remove console statements

File: src/pages/Auth/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                type="text"
+                className="input-feild"
+                value={loginForm.email}
+                placeholder="Enter your email here"
+                onChange={(e)=>setLoginForm({...loginForm,email:e.target.value})}
+                required
+              />
+            </div>
+            <div className="auth-input flex-row">
+              <label for="" className="input-label"
```
Comment (rohit8686) [inline src/pages/Auth/Login.jsx:31]: do not leave "for" attribute empty

**Review by rohit8686 — APPROVED** [review]
I have mentioned few suggestions, other than that code is nice and modular and folder structure is also good

**Review by anuj-kumary — APPROVED** [review]
Code is looking good and also liked how you used a naming convention for function and classname

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        });
+        var response = await loginServices(
+          "adarshbalika@gmail.com",
+          "adarshBalika123"
+        );
+      } else var response = await loginServices(email, password);
+
+      if (response.status === 201 || response.status === 200) {
+        localStorage.setItem(
+          "auth",
```
Comment (vinc3nati) [inline src/context/auth-context.js:28]: Very modular and clean code, but as you see here you are reusing the key "auth" everytime in a hardcoded way, better just put it in a const variable and make it global to access

**Review by vinc3nati — APPROVED** [review]

**Pull request merged**